hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -309,6 +310,16 @@ public WALEntryBatch poll(long timeout) throws InterruptedException {
     return entryBatchQueue.poll(timeout, TimeUnit.MILLISECONDS);
   }
 
+  public void clearWALEntryBatch() {
+    entryBatchQueue.forEach(w -> {
+      entryBatchQueue.remove(w);
+      w.getWalEntries().forEach(e -> {
```

Comment:
What about summing the total size to decrement and then making the one call to `totalBufferUsed`? Guessing that might be micro-optimized faster.

Reply:
Makes sense, coming soon...